### ISSUE TITLE
[lsp-server] 🐞 rejectUnauthorized doesn't work as advertized

### ISSUE THREAD

**Issue by TimMensch** [graphql/graphiql#3596]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If you have a self-signed certificate on your GraphQL server, the "rejectUnauthorized" flag is supposed to allow the VS Code GraphQL Language Feature Support extension to accept a self-signed certificate.

Ideally, the option would allow us to select a root certificate to trust, but at the very least the documented option should allow self-signed certificates to work.

At present, every single interaction results in:

```
[Error - 8:48:26 PM] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: unable to verify the first certificate
  Code: -32603 
```

### Expected Behavior

It should work with a self-signed cert. 'nuff said.

### Steps To Reproduce

Same setup as this older bug:

https://github.com/graphql/graphiql/issues/451

In other words:

1. Create a self-signed certificate.
2. Serve GraphQL from that server.
3. Note that the GraphQL language server refuses to connect.

The VS Code extension has an option "vscode-graphql.rejectUnauthorized" that is supposed to set the language server to accept self-signed certificates. It doesn't work.

### Environment

* LSP Server Version: Heck if I know. The output window has nothing in it resembling a version number. Whatever comes with the latest version of the VS Code extension. 
* OS: Windows 11
* LSP Client: vscode (graphql.vscode-graphql-0.9.3)

### Anything else?

I also tried the prerelease version. No difference in behavior.
